cmd/modelcmd/base.go

```diff
@@ -249,6 +249,9 @@ func (c *CommandBase) NewAPIRoot(
 	if redirErr, ok := errors.Cause(err).(*api.RedirectError); ok {
 		return nil, newModelMigratedError(store, modelName, redirErr)
 	}
+	if _, ok := errors.Cause(err).(*juju.NoAddressesError); ok {
```

Comment:
We normally also define a IsNoAddressesError() helper to use
And the test becomes c.Assert(err, jc.Satisfies, juju.IsNoAddressesError)
We could then get away with not exporting the error, just the helper